src/coreComponents/physicsSolvers/contact/LagrangianContactSolver.cpp

```diff
@@ -217,7 +217,10 @@ void LagrangianContactSolver::implicitStepComplete( real64 const & time_n,
                                                     real64 const & dt,
                                                     DomainPartition & domain )
 {
-  m_solidSolver->implicitStepComplete( time_n, dt, domain );
+  if( m_setupSolidSolverDofs )
+  {
+    m_solidSolver->implicitStepComplete( time_n, dt, domain );
+  }
```

Comment:
After this PR is merged, we could explore ways to avoid the need for this flag. The need for this flag may indicate the current `LagrangianContactSolver` does too much work and should be broken down into two solvers (one coupled solver and one "single-physics" solver). Disclaimer: I don't know the existing `LagrangianContactSolver` very well, so I may be missing something.

**LagrangianContactSolver as a coupled solver**
The coupled solver would be something like that:
```
class LagrangianContactSolver : public CoupledSolver< SolidMechanicsLagrangianFEM, ContactSolver >
```
and would just be a driver for `setupDofs`, `implicitStepSetup`, `implicitStepComplete`, `applySystemSolution`, `applyBoundaryConditions` (functions where the current solver needs access to `m_solidSolver`). It would not contain any assembly functions.

**New single-physics solver**
The single-physics solver `ContactSolver` (or a better name) would have the assembly functions `assembleForceResidualDerivativeWrtTraction`, `assembleTractionResidualDerivativeWrtDisplacementAndTraction`, and `assembleStabilization` because these functions do not require access to the pointer to `m_solidSolver`. This single-physics solver would not make sense on its own, but to me this is ok.

**SinglePhasePoromechanicsConformingFractures**
Finally, in the new coupled solver added in this PR, we would have:
```
class SinglePhasePoromechanicsConformingFractures : public CoupledSolver< SinglePhasePoromechanicsSolver, ContactSolver >
```
and we would have the guarantee that `SolidMechanicsLagrangianFEM` can only be accessed by doing `poromechanicsSolver()->solidMechanicsSolver()`. So, with this approach, 1) we would not need `m_setupSolidSolverDofs` any more, and 2) we could still easily reuse the contact solver assembly functions.

Reply:
yes, I think this is doable. The only thing I am not too sure how to do is how to have a contact solver that is meaningful by itself. Otherwise we will just need a way to make sure a user cannot try to run it without a `SolidSolver`.